.github/workflows/trigger-gitlab-pipeline.yml

```diff
@@ -5,7 +5,40 @@ on:
       - main  # This ensures the workflow triggers on pushes to the main branch, including merges
 
 jobs:
-  trigger-gitlab:
+  trigger-and-check-gitlab:
     runs-on: ubuntu-latest
     steps:
-      - run: curl --request POST --form token=${{secrets.GITLAB_TOKEN}} --form ref=main "https://gitlab.com/api/v4/projects/35484791/trigger/pipeline"
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Trigger GitLab Pipeline
+        id: trigger
+        run: |
+          response=$(curl --request POST \
+                           --form token=${{ secrets.GITLAB_TRIGGER_TOKEN }} \
+                           --form ref=main \
+                           "https://gitlab.com/api/v4/projects/35484791/trigger/pipeline")
+          echo "::set-output name=pipeline_id::$(echo $response | jq -r '.id')"
+      
+      - name: Poll Pipeline Status
+        run: |
+          status=""
+          while [[ "$status" != "success" && "$status" != "failed" ]]; do
+            result=$(curl "https://gitlab.com/api/v4/projects/35484791/pipelines/${{ steps.trigger.outputs.pipeline_id }}")
+            status=$(echo $result | jq -r '.status')
+            echo "Pipeline status: $status"
+            if [[ "$status" == "failed" ]]; then
```

Comment:
question: I don't know how many status this response might return, I guess is `success`, `pending` and `failed`, but I'd like to check if there is others that we might handle as an error as well, e.g. `cancelled`.

Reply:
Wonderful feedback!

Will validate states as I'm only aware of the three mentioned (success, failed, pending). It's very possible there could be additional, as you mentioned.

Reply:
Looks like these are the states a given pipeline in GitLab can reflect per https://docs.gitlab.com/ee/api/pipelines.html

created, waiting_for_resource, preparing, pending, running, success, failed, canceled, skipped, manual, scheduled

Reply:
I thin should be enough to check `failed` and `canceled` here

Reply:
I went ahead and committed the changes we talked about. I think this covers it?